File: sdk/lib/bot.lua
return {
    send = function (s)
        local len = #s
        if len >= 65536 * 256 then
            io.stderr('Message too long!')
            return
        end
        local a, b, c = len % 256, math.floor(len / 256) % 256, math.floor(len / 65536)
        local t = string.char(a, b, c)
        io.stdout:write(t)
        io.stdout:write(s)
        io.stdout:flush()
    end,

    recv = function ()
        local s = io.stdin:read(3)
        local a, b, c = string.byte(s, 1, 3)
        local t = io.stdin:read(a + (b * 256) + (c * 65536))
        return t
    end,

    log = function (s)
        io.stderr:write(s)
        io.stderr:flush()
    end
}

File: sdk/lib/bot.py
import sys

class Bot:
    @staticmethod
    def send(s):
        l = len(s)
        if l >= (1 << 24):
            sys.stderr.write('Message too long!')
            return
        sys.stdout.write(chr(l & 0xff))
        sys.stdout.write(chr((l >> 8) & 0xff))
        sys.stdout.write(chr(l >> 16))
        sys.stdout.write(s)
        sys.stdout.flush()

    @staticmethod
    def recv():
        s = sys.stdin.buffer.read(3)
        l = s[0] + (s[1] << 8) + (s[2] << 16)
        return sys.stdin.buffer.read(l).decode()

    @staticmethod
    def log(s):
        sys.stderr.write(s)
        sys.stderr.flush()
